airflow/www/templates/airflow/trigger.html

```diff
@@ -86,8 +86,9 @@
         {% endfor -%}
       </select>
     {% elif form_details.schema and "array" in form_details.schema.type %}
-      {% if "items" in form_details.schema and form_details.schema.items %}
-      <textarea class="form-control" name="element_{{ form_key }}" id="element_{{ form_key }}" valuetype="advancedarray" rows="6"
+      {% if "items" in form_details.schema and form_details.schema["items"] %}
+      <textarea class="form-control" name="element_{{ form_key }}" id="element_{{ form_key }}" rows="6"
+        {%- if "string" in form_details.schema["items"]["type"] %} valuetype="array"{% else %} valuetype="advancedarray"{% endif -%}
```

Comment:
With the proposed approach you generate a redundancy to lines 112-119. How about if you check the negative option in line 89 such that `"string" not in form_details.schema["items"]["type"]` - then the advancedarray is only rendered for all non-string editions.

Reply:
You are right, we have some redundancy here. Once the formatting is solved in JS and once in the template, what is the preferred way here?

Yes the `advancedarray` is rendered for all other types. You can have for example an array of objects with format validation in it. The default element of an array type array is the textarea, when any different type than string is specified the CodeMirror element is rendered.

Reply:
I just realized that introduces the redundancy unnecessarily, thanks for the feedback. I just updated my solution. It follows the same behavior.